juju/juju
# WIP : Initial changes for bundle export

## Please provide the following details to expedite Pull Request review:
*Please replace with a description about why this change is needed?*
This PR addresses code changes to support "export bundle" functionality in juju.
Currently this feature exists currently in juju-gui. 
This export the current model into a bundle in yaml format which can then be used to deploy locally.

## QA steps
*Please replace with how we can verify that the change works?*
Yet to implement the CLI part. The idea is to implement a new command as below
'juju export-bundle [modelname]
default model or the model which is currently chosen is picked up if the model name is not specified.

## Documentation changes
*Please replace with any notes about how it affects current user workflow? CLI? API?* 
** to be filled in later **
## Bug reference
*Please add a link to any bugs that this change is related to.*
N/A
